packages/playback-core/src/index.ts

```diff
@@ -30,12 +30,34 @@ export const generatePlayerInitTime = () => {
   return mux.utils.now();
 };
 
+export const generateUUID = mux.utils.generateUUID;
+
 export const toMuxVideoURL = (playbackId?: string, { domain = MUX_VIDEO_DOMAIN } = {}) => {
   if (!playbackId) return undefined;
   const [idPart, queryPart = ''] = toPlaybackIdParts(playbackId);
   return `https://stream.${domain}/${idPart}.m3u8${queryPart}`;
 };
 
+const toPlaybackIdFromParameterized = (playbackIdWithParams: string | undefined) => {
+  if (!playbackIdWithParams) return undefined;
+  const [playbackId] = playbackIdWithParams.split('?');
+  // `|| undefined` is here to handle potential invalid cases
+  return playbackId || undefined;
+};
+
+const toPlaybackIdFromSrc = (src: string | undefined) => {
+  if (!src || !src.startsWith('https://stream.')) return undefined;
```

Comment:
what if a custom domain is used?

Reply:
Should this use `isMuxVideoSrc`?

Reply:
Re: custom domains - I'm pretty confident those *should* be fine based on how they work, but I should confirm that there won't be exceptions. (tl;dr - custom video domains are still "ours", aka not a proxy, just a CNAME mapping)
Re: `isMuxVideoSrc` - This function isn't called directly, but instead used by `toVideoId`, which does check `isMuxVideoSrc` (which includes custom domains, fwiw)

Reply:
I was a little confused too. custom domains always still have the `https://stream.` prefix then, correct?

Reply:
In a conversation elsewhere, it sounds like that's a yes.

Reply:
yessir(s)!